assignment4/disjointset.py

```diff
@@ -0,0 +1,49 @@
+class DisjointSet:
+    """Implementation of a Disjoint Set."""
+
+    def __init__(self):
+        """
+        Fields:
+            parents: link to the parent (header member) of the set
+            rank:    helps to determine how to union two sets
+                     (heuristic to improve time complexity of unionSet)
+        As well as I want to use 1-based numeration of set, `parent` and `rank`
+        contain first extra element.
+        """
+        self.parent = [0]
```

Comment:
Nit: you should make these `__parent`, `__rank`, and `__set_count`.

This is so the fields are marked as private. Users of this class should not be able to mess around with the internals then.